File: lista 15questoes/15 questoes em python/att05.py
#5. Polimorfismo Utilize polimorfismo para criar uma função que receba uma lista de objetos Animal e chame o método som de cada um, demonstrando comportamento diferente para cada subclasse. 

class Animal:
    def __init__(self,nome):
        self.nome = nome

    def som(self):
        pass

class Cachorro(Animal):
    def som(self):
        print(f"{self.nome} faz: Au Au")

class Gato(Animal):
    def som(self):
        print(f"{self.nome} faz: miau miau")


def fazer_barulho(animais):
    for animal in animais:
        animal.som()



cachorro1 = Cachorro("scooby")
gato1 = Gato("Mu")

lista_animais = [cachorro1,gato1]

fazer_barulho(lista_animais)

File: lista 15questoes/15 questoes em java/att05.java
class Animal {
    String nome;

    Animal(String nome) {
        this.nome = nome;
    }

    void som() {}
}

class Cachorro extends Animal {
    Cachorro(String nome) {
        super(nome);
    }

    void som() {
        System.out.println(nome + " faz: Au Au");
    }
}

class Gato extends Animal {
    Gato(String nome) {
        super(nome);
    }

    void som() {
        System.out.println(nome + " faz: Miau Miau");
    }
}

class Main {
    static void fazerBarulho(Animal[] animais) {
        for (Animal animal : animais) {
            animal.som();
        }
    }

    public static void main(String[] args) {
        Cachorro cachorro1 = new Cachorro("Scooby");
        Gato gato1 = new Gato("Mu");
        Animal[] listaAnimais = {cachorro1, gato1};
        fazerBarulho(listaAnimais);
    }
}
